camel/toolkits/openbb_toolkit.py

```diff
@@ -0,0 +1,1126 @@
+            return data.to_df() if return_type == "df" else data.results
+        except Exception as e:
+            msg = f"Failed to get stock quote for {symbol}: {e}"
+            print(msg)
+            return pd.DataFrame() if return_type == "df" else {}
+
+    def get_historical_data(
+        self,
+        symbol: str,
+        asset_type: Literal[
+            "equity", "index", "currency", "crypto", "future", "option"
+        ] = "equity",
+        start_date: Optional[str] = None,
+        end_date: Optional[str] = None,
+        interval: Literal["1m", "1h", "1d", "1W", "1M"] = "1d",
+        provider: Optional[str] = None,
+        return_type: Literal["df", "raw"] = "df",
+    ) -> Union[pd.DataFrame, Dict]:
+        """Get historical price data for various financial instruments.
+
+        Args:
+            symbol (str): Symbol to get data for
+            asset_type (Literal): Type of asset
+            start_date (Optional[str]): Start date
+            end_date (Optional[str]): End date
+            interval (Literal): Data interval
+            provider (Optional[str]): Data provider
+            return_type (Literal): Return format
+
+        Returns:
+            Union[pd.DataFrame, Dict]: Historical price data
+        """
+        if provider is None:
+            provider_map = {
+                "equity": "yfinance",
+                "index": "cboe",
+                "currency": "polygon",
+                "crypto": "polygon",
+                "future": "fmp",
+                "option": "polygon",
+            }
+            provider = provider_map[asset_type]
+
+        formatted_symbol = self._format_symbol(symbol, asset_type, provider)
+
+        if asset_type == "equity":
+            data = self.client.equity.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        elif asset_type == "index":
+            data = self.client.index.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        elif asset_type == "currency":
+            data = self.client.currency.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        elif asset_type == "crypto":
+            data = self.client.crypto.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        else:
+            data = self.client.equity.price.historical(
```

Comment:
Just want to double check here, so the equity and options share the same mechanism?

Reply:
Thanks @willshang76 , just update the code to have a more concise and clearer code which matches the examples OpenBB gave.